packages/editor/src/client/routes/articleEditor.tsx

```diff
@@ -482,6 +491,19 @@ export function ArticleEditor({id}: ArticleEditorProps) {
                   )}
                 </div>
               }
+              rightChildren={
+                <IconButtonLink
+                  disabled={hasChanged || !id}
+                  style={{marginTop: '4px'}}
+                  size={'lg'}
+                  icon={<Icon icon="eye" />}
+                  onClick={async e => {
+                    await previewLinkRefetch({id: id, hours: 1})
+                    window.open(data?.articlePreviewLink || '')
```

Comment:
Also if `previewLinkRefetch()` doesn't return anything you shouldn't open a new tab. Here you can just do nothing if the query doesn't return a link,